### PR TITLE
Fix issue 90 - empty manifest config support on Windows

### PR DESCRIPTION
**Pull request by sunnycarter**
Fix for https://github.com/oras-project/oras-py/issues/90

Tested with repro on Windows and Linux.

I'm not quite sure what to do with the CHANGELOG.md I'm afraid so please give guidance.

### CODE REVIEW

File: oras/oci.py [hunk 1]
```diff
@@ -128,7 +129,10 @@ def ManifestConfig(
     """
     # Create an empty config if we don't have one
     if not path or not os.path.exists(path):
-        path = "/dev/null"
+        if platform.system() == "Windows":
+            path = "nul"
+        else:
+            path = "/dev/null"
```
Comment (vsoch) [inline oras/oci.py:135]: Here you can do:

```python
path = "nul" if platform.system() == "Windows" else "/dev/null" 
```

File: oras/provider.py [hunk 2]
```diff
@@ -430,7 +431,10 @@ def download_blob(
         # Allow an empty layer to fail and return /dev/null
         except Exception as e:
             if digest == oras.defaults.blank_hash:
-                return "/dev/null"
+                if platform.system() == "Windows":
```
Comment (vsoch) [inline oras/provider.py:434]: If we are doing this twice - maybe write a util function instead?

File: oras/oci.py (hunk 1 above)
Comment (sunnycarter) [reply oras/oci.py]: I found a better way to fix with os.devnull

File: oras/provider.py (hunk 2 above)
Comment (sunnycarter) [reply oras/provider.py]: Found that os.devnull exists

**Comment by sunnycarter** [comment]
Thanks for the review, the nod to write a utility function led me to discover there's an inbuilt one

**Comment by vsoch** [comment]
> Thanks for the review, the nod to write a utility function led me to discover there's an inbuilt one

Nice! And good intuition to look for that too!

For the DCO you can click on "Details" to learn more about that - if I remember it's just an electronic page you need to agree to and then you are good. It would only be done once! The ORAS family of projects is under the CNCF umbrella so we follow these policies that are enforced there.

**Comment by vsoch** [comment]
> I'm not quite sure what to do with the CHANGELOG.md I'm afraid so please give guidance.

For this, you'll want to increment the version to 0.1.18 here: https://github.com/oras-project/oras-py/blob/209c9b98043a00d1b04789cc2967ca7021dc5b2e/oras/version.py#L5

And then make a note (with that corresponding version) on a newline above 0.1.17 here  https://github.com/oras-project/oras-py/blob/main/CHANGELOG.md

**Comment by sunnycarter** [comment]
Thanks for looking at this, I've now signed all commits to fulfil the DCO, and pushed a new commit with a patch version bump and changelog update.

**Review by vsoch — APPROVED** [review]

**Pull request merged**